native/src/main/java/io/ballerina/stdlib/etl/utils/CommonUtils.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    public static BArray initializeNestedBArray(BTypedesc type, int size) {
+        ArrayType arrayType = TypeCreator.createArrayType(TypeUtils.getReferredType(type.getDescribingType()));
+        BArray nestedArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(arrayType, size));
+        return nestedArray;
+    }
+
+    public static BMap<BString, Object> initializeBMap(BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        return ValueCreator.createRecordValue(
+                TypeCreator.createRecordType(describingType.getName(), describingType.getPackage(),
+                        describingType.getFlags(), false, 0));
+    }
+
+    public static BMap<BString, Object> copyBMap(BMap<BString, Object> source, BTypedesc targetType) {
+        BMap<BString, Object> target = initializeBMap(targetType);
+        for (BString key : source.getKeys()) {
+            target.put(key, source.get(key));
+        }
+        return target;
+    }
+
+    public static BArray convertJSONToBArray(Object source, BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        return (BArray) JsonUtils.convertJSON(source, TypeCreator.createArrayType(describingType));
+    }
+
+    public static BArray convertJSONToNestedBArray(Object source, BTypedesc type) {
+        ArrayType arrayType = TypeCreator.createArrayType(TypeUtils.getReferredType(type.getDescribingType()));
+        return (BArray) JsonUtils.convertJSON(source, TypeCreator.createArrayType(arrayType));
+    }
+
+    public static Object convertJSONToRecord(Object source, BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        return JsonUtils.convertJSON(source, describingType);
+    }
+
+    public static BString[] getFields(BTypedesc type) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        StructureType structType = (StructureType) describingType;
+        Map<String, Field> fields = structType.getFields();
+        BString[] fieldNames = new BString[fields.size()];
+        int i = 0;
+        for (Map.Entry<String, Field> entry : fields.entrySet()) {
+            fieldNames[i++] = StringUtils.fromString(entry.getKey());
+        }
+        return fieldNames;
+    }
+
+    public static boolean isFieldExist(BArray dataset, BString fieldName) {
+        Type describingType = TypeUtils.getReferredType(dataset.getElementType());
+        StructureType structType = (StructureType) describingType;
+        Map<String, Field> fields = structType.getFields();
+        return fields.containsKey(fieldName.getValue());
+    }
+
+    public static String getFieldType(BTypedesc type, BString fieldName) {
+        Type describingType = TypeUtils.getReferredType(type.getDescribingType());
+        StructureType structType = (StructureType) describingType;
```

Comment:
This can throw a `TypeCastingException`. We should check the type tag before casting. Check similar places.